### PR TITLE
chore(Jenkinsfile) switch to pipeline-library for terraform tasks

### PR DESCRIPTION
**Pull request by dduportal**
This PR switches the current pipeline to use the new pipeline-library introduced in https://github.com/jenkins-infra/pipeline-library/pull/258.

It features a git submodule with the reusable terraform tooling stored at https://github.com/jenkins-infra/shared-tools/tree/main/terraform

Documentation updated, and pipeline tests (unit + PR here)

### CODE REVIEW

File: Jenkinsfile_k8s [hunk 1]
```diff
@@ -1,141 +1,34 @@
...
+            file(variable: 'BACKEND_CONFIG_FILE', credentialsId: 'production-terraform-aws-backend-config'),
+          ],
+        )
+      }
+    }
+  },
+  'updatecli': {
+    withCredentials([string(credentialsId: 'updatecli-github-token', variable: 'UPDATECLI_GITHUB_TOKEN')]) {
+      updatecli(action: 'diff')
+      if (env.BRANCH_NAME == 'main') {
```
Comment (lemeurherve) [inline Jenkinsfile_k8s:29]:
```suggestion
      if (env.BRANCH_IS_PRIMARY) {
```

**Review by lemeurherve — APPROVED** [review]
🎉

File: README.adoc [hunk 2]
```diff
@@ -1,62 +1,50 @@
...
-** Run `make lint` to statically lint the project resources. It fails if you didn't.
-** Run `make lint-tests` to statically lint the test resources. It fails if you didn't.
-** Run `make prepare` to initialize the local Terraform project (e.g. generates `.terraform` and retrieve dependencies, using the S3 backend)
-** Run `make validate` to run a static analysis of the project (both Terraform and terratest code)
-** Run `make test` to execute the test harness on the project (uses the terratest code in `./tests/`)
-** Run `make plan` to prints the changes planned to be applied on the infrastructure (see this step as a "dry run" of `make deploy`)
-** Run `make deploy` to apply the changes to the infrastructure (implies terraform apply with no approval)
+[source,bash]
+----
+make --directory=.shared-tools/terraform help
```
[the remark this answers is not in the document]
Comment (dduportal) [reply README.adoc:33]: how would you use it? Except `make --directory="${SHARED_TOOLS_DIRECTORY}" help` ?

[the remark this answers is not in the document]
Comment (lemeurherve) [reply README.adoc:33]: Nevermind, mistake from my part.

**Review by lemeurherve — APPROVED** [review]

**Pull request merged**